--- a/configure.wrf
+++ b/configure.wrf
@@ -233,7 +233,7 @@
 LIB             =    $(LIB_BUNDLED) $(LIB_EXTERNAL) $(LIB_LOCAL) $(LIB_WRF_HYDRO) 
 LDFLAGS         =    $(OMP) $(FCFLAGS) $(LDFLAGS_LOCAL) 
 ENVCOMPDEFS     =    
 WRF_CHEM	=	0 
-CPPFLAGS        =    $(ARCHFLAGS) $(ENVCOMPDEFS) -I$(LIBINCLUDE) $(TRADFLAG) 
+CPPFLAGS        =    $(ARCHFLAGS) $(ENVCOMPDEFS) -DWRF_SAVEPOINT -I$(LIBINCLUDE) $(TRADFLAG) 
 NETCDFPATH      =    /home/enric/src/canairy_meteo/Gen2/artifacts/wrf_gpu_libs
 HDF5PATH        =    /home/enric/src/canairy_meteo/Gen2/artifacts/wrf_gpu_libs
 WRFPLUSPATH     =    
